includes/embeds/class-amp-pinterest-embed-handler.php

```diff
@@ -12,7 +12,7 @@
  */
 class AMP_Pinterest_Embed_Handler extends AMP_Base_Embed_Handler {
 
-	const URL_PATTERN = '#https?://(www\.)?pinterest\.com/pin/.*#i';
+	const URL_PATTERN = '#https?://(?:www\.)?(?:[a-z]{2}\.)?pinterest\.[a-z.]+/([^/]+)(/[^/]+)?#i';
```

Comment:
@pierlon This seems too permissive for AMP. The `amp-pinterest` component [only supports the `pin` widget](https://github.com/ampproject/amphtml/tree/b6313e372fdd1298928e2417dcc616b03288e051/extensions/amp-pinterest/0.1), but not the [two other](https://developers.pinterest.com/tools/widget-builder) existing widgets: `board` and `user`.

This pattern will, however, match all those URLs: pin, board and user – and in turn will display an empty widget when the user tries to embed a board or a user profile to the page. I believe the original path (`/pin/.*`) should still be used.

Also, is there any concern with using the wildcard to match the tld?

I'm not sure if there is an attack vector, but I'm wondering if someone could snatch an obscure `pinterest.<something>` domain and trick users to adding that string to their posts, which in turn would trigger an oEmbed request and potentially render malicious markup on the website. I don't believe that could happen in AMP context, but I'm also not 100% sure if this embed handler only runs in AMP context.

Reply:
Those are some good points you made there @dero. The removal of the `/pin/` path from the regex was an oversight and will be added back to the pattern.

As for the potential TLD attack vector you mentioned, I don't think it would be an issue because all the AMP component really needs is the [pin ID from the supplied URL](https://github.com/ampproject/amphtml/blob/b6313e372fdd1298928e2417dcc616b03288e051/extensions/amp-pinterest/0.1/pin-widget.js#L330). I'm not familar with the list of TLDs Pinterest owns (or would own in the future), so I don't want to add any more specificity to the regex pattern that would unknowingly reject genuine Pinterest URLs.

Reply:
See https://github.com/ampproject/amp-wp/pull/5299/commits/27b173be3014c2ce97c117abdcceaad23de0af4b for the updated regex and additional test you recommended in https://github.com/ampproject/amp-wp/pull/5299#discussion_r480871426.